src/Maths/math.vector.ts

```diff
@@ -509,6 +509,44 @@ export class Vector2 {
         return new Vector2(x, y);
     }
 
+    /**
+    * Moves a value current towards target.
+    *
+    * This is essentially the same as Vector2.Lerp but instead the function will ensure that the speed never exceeds maxDelta.

```

Comment:
I believe the documentation is confusing - maxDistance is not speed, it is, as the name suggests, the maximum distance the result will be from the origin